### PR TITLE
[JSON-API] Log templateId & choice name (if present) on command submissions in the json API

### LINKED ISSUE #9983: Log template id and (if present) choice name on command submissions in the JSON API
When submitting a command to the ledger from the JSON API we should also include the template id and (for exercise/create-and-exercise) the choice name. Probably wouldn’t throw it in the context but include it in the string just like we do for the type of command.

### PR DESCRIPTION
**Pull request by realvictorprm**
fixes #9983

changelog_begin

- [JSON API] The template id & choice name (if present) are now logged on command submissions in the Json API (at trace level)

changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by stefanobaghino-da — COMMENTED** [review]
In general when logging some form of data it's probably better to put it in the logging context so that it's available as part of the structured part of the log entry.

**Comment by cocreature** [comment]
> In general when logging some form of data it's probably better to put it in the logging context so that it's available as part of the structured part of the log entry.

I originally suggested not putting it in the logging context since it seemed like that might be getting a bit out of hand especially for things that are only logged once. But you have much more experience in that area so happy to follow your guidance and move it to the logging context if you think it makes sense.

**Comment by stefanobaghino-da** [comment]
> > In general when logging some form of data it's probably better to put it in the logging context so that it's available as part of the structured part of the log entry.
> 
> I originally suggested not putting it in the logging context since it seemed like that might be getting a bit out of hand especially for things that are only logged once. But you have much more experience in that area so happy to follow your guidance and move it to the logging context if you think it makes sense.

The idea would be that one should be able to easily process logs in structured form without parsing the message, that's why I was suggesting that. However I don't have a strong opinion if for any reason we prefer to have this in the message.

**Comment by realvictorprm** [comment]
OK folks, I'm also in favour of including it in the context, because I also had my thoughts about the usability here.

I think generally it is actually best to include everything in the context which later most likely is going to be matched for during debugging/investigating an issue.

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala [hunk 1]
```diff
@@ -40,65 +42,88 @@ class CommandService(
...
   )(implicit
       lc: LoggingContextOf[InstanceUUID with RequestID]
-  ): Future[Error \/ ExerciseResponse[lav1.value.Value]] = {
-    logger.trace("sending exercise command to ledger")
-    val command = exerciseCommand(input)
-    val request = submitAndWaitRequest(jwtPayload, input.meta, command, "exercise")
+  ): Future[Error \/ ExerciseResponse[lav1.value.Value]] =
+    withTemplateChoiceLoggingContext(input.reference.fold(_._1, _._1), input.choice).run {
+      implicit lc =>
+        logger.trace(s"sending exercise command to ledger")
```
Comment (stefanobaghino-da) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala:91]:
```suggestion
        logger.trace("sending exercise command to ledger")
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala [hunk 2]
```diff
@@ -40,65 +42,88 @@ class CommandService(
...
-      response <- logResult(
-        Symbol("createAndExercise"),
-        submitAndWaitForTransactionTree(jwt, request),
-      )
-      exerciseResult <- either(exerciseResult(response))
-      contracts <- either(contracts(response))
-    } yield ExerciseResponse(exerciseResult, contracts)
+  ): Future[Error \/ ExerciseResponse[lav1.value.Value]] =
+    withTemplateChoiceLoggingContext(input.templateId, input.choice).run { implicit lc =>
+      logger.trace(s"sending create and exercise command to ledger")
```
Comment (stefanobaghino-da) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala:113]:
```suggestion
      logger.trace("sending create and exercise command to ledger")
```

**Comment by realvictorprm** [comment]
Side note, I try to only force push if necessary, which currently is limited to rebasing.

**Review by stefanobaghino-da — APPROVED** [review]

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala (hunk 1 above)
Comment (realvictorprm) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala]: great, thanks!

**Review by cocreature — APPROVED** [review]
tHANKS!

**Pull request merged**